addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,422 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+from odoo import models, fields, api, _
+from odoo.tools import float_is_zero
+from odoo.exceptions import RedirectWarning, UserError
+import logging
+import requests
+
+_logger = logging.getLogger(__name__)
+DEFAULT_ENDPOINT = "https://gw.fbr.gov.pk/DigitalInvoicing/v1/PostInvoiceData_v1"
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = "account.edi.format"
+
+    def _is_enabled_by_default_on_journal(self, journal):
+        self.ensure_one()
+        if self.code == "pk_einvoice_1_3":
+            return False
+        return super()._is_enabled_by_default_on_journal(journal)
+
+    def _get_l10n_pk_sales_group_ids(self, company_id):
+        ChartTemplate = self.env['account.chart.template'].with_company(company_id)
+        return (
+            ChartTemplate.ref('tax_group_pk_0')
+            + ChartTemplate.ref('tax_group_pk_1')
+            + ChartTemplate.ref('tax_group_pk_2')
+            + ChartTemplate.ref('tax_group_pk_3')
+            + ChartTemplate.ref('tax_group_pk_5')
+            + ChartTemplate.ref('tax_group_pk_8')
+            + ChartTemplate.ref('tax_group_pk_10')
+            + ChartTemplate.ref('tax_group_pk_13')
+            + ChartTemplate.ref('tax_group_pk_15')
+            + ChartTemplate.ref('tax_group_pk_16')
+            + ChartTemplate.ref('tax_group_pk_17')
+            + ChartTemplate.ref('tax_group_pk_195')
+        ).ids
```

Comment:
Even better version, it uses the ORM cache and probably won't even make a single query.

```py
In [1]: def _get_l10n_pk_sales_groups(company_id):
   ...:     return self.env['account.tax.group'].browse([
   ...:         sale_group_id
   ...:         for ref in (
   ...:             'tax_group_pk_0',
   ...:             'tax_group_pk_1',
   ...:             'tax_group_pk_2',
   ...:             'tax_group_pk_3',
   ...:             'tax_group_pk_5',
   ...:             'tax_group_pk_8',
   ...:             'tax_group_pk_10',
   ...:             'tax_group_pk_13',
   ...:             'tax_group_pk_15',
   ...:             'tax_group_pk_16',
   ...:             'tax_group_pk_17',
   ...:             'tax_group_pk_195',
   ...:         )
   ...:         if (
   ...:             (full_ref := f"account.{company_id.id}_{ref}")
   ...:             and (sale_group_id := self.env['ir.model.data']._xmlid_to_res_model_res_id(full_ref)[1])
   ...:         )
   ...:     ])

In [2]: _get_l10n_pk_sales_groups(self.env['res.company'].browse([2]))
Out[2]: account.tax.group(3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14)

```

Reply:
Insightful 🧠...Well we might adapt it in `l10n_in` too because we use the similar mechanism after this PR merges. @lordkrandel Is there something we can do it more generic way to do it for `account`? i.e. fetching a list of xml_ids of same model (well may be let's just keep things simple as of now and focus on this PR, I have been thinking of too far)